qi-doc/scribblings/list-operations.scrbl

```diff
@@ -0,0 +1,76 @@
+#lang scribble/doc
+@require[scribble/manual
+	 (for-label racket/list
+	 	    racket/base)]
+
+@title{List Operations}
+
+@defmodule[qi/list]
+
+This module defines bindings that can leverage stream fusion /
+deforestation optimization when found in succession within a
+flow. When not part of optimized flow, their behavior is identical to
+the bindings of the same name from @racketmodname[racket/base] and
+@racketmodname[racket/list].
+
+The bindings are categorized based on their intended usage inside the
+deforested pipeline.
+
+@section{Producers}
+
+@defproc*[(((range (end real?)) list?)
+           ((range (start real?) (end real?) (step real? 1)) list?))]{
+
+Deforestable version of @racket[range] from @racketmodname[racket/list].
```

Comment:
This repeatable pattern could be extracted:
```
(define-syntax-rule (deforestable what where)
  @para{Deforestable version of @racket[@what] from @racketmodname[@where]})
```
(or something similar: I'm not actually sure how to get the right stuff in the
racket and racketmodname forms in Scribble, since they are syntax).

Reply:
Documentation improvements PR too.